src/components/Avatar.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {useEffect, useState} from 'react';
+import {StyleProp, View, ViewStyle} from 'react-native';
+import {ValueOf} from 'type-fest';
+import useNetwork from '@hooks/useNetwork';
+import * as ReportUtils from '@libs/ReportUtils';
+import {AvatarSource} from '@libs/UserUtils';
+import * as StyleUtils from '@styles/StyleUtils';
+import useTheme from '@styles/themes/useTheme';
+import useThemeStyles from '@styles/useThemeStyles';
+import CONST from '@src/CONST';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import Image from './Image';
+
+type AvatarProps = {
+    /** Source for the avatar. Can be a URL or an icon. */
+    source?: AvatarSource;
+
+    /** Extra styles to pass to Image */
+    imageStyles?: ViewStyle[];
+
+    /** Additional styles to pass to Icon */
+    iconAdditionalStyles?: StyleProp<ViewStyle>;
+
+    /** Extra styles to pass to View wrapper */
+    containerStyles?: StyleProp<ViewStyle>;
+
+    /** Set the size of Avatar */
+    size?: ValueOf<typeof CONST.AVATAR_SIZE>;
+
+    /**
+     * The fill color for the icon. Can be hex, rgb, rgba, or valid react-native named color such as 'red' or 'blue'
+     * If the avatar is type === workspace, this fill color will be ignored and decided based on the name prop.
+     */
+    fill?: string;
+
+    /** A fallback avatar icon to display when there is an error on loading avatar from remote URL.
+     * If the avatar is type === workspace, this fallback icon will be ignored and decided based on the name prop.
+     */
+    fallbackIcon?: AvatarSource;
+
+    /** Denotes whether it is an avatar or a workspace avatar */
+    type?: typeof CONST.ICON_TYPE_AVATAR | typeof CONST.ICON_TYPE_WORKSPACE;
```

Comment:
I think I already saw similar type like this , maybe we can extract it to some seprate type?

Reply:
That's true, I've seen in repeated in other components also. I can create a separate type and update it where needed. @blazejkustra, what do you think about it?

Reply:
Sure, go ahead! You can create it in `src/types/onyx/OnyxCommon`